charts/dex/templates/networkpolicy.yaml

```
@@ -0,0 +1,38 @@
+{{- if .Values.networkPolicy.enabled }}
+{{- if semverCompare "<1.7-0" .Capabilities.KubeVersion.GitVersion -}}
+apiVersion: extensions/v1beta1
+{{- else -}}
+apiVersion: networking.k8s.io/v1
+{{- end }}
+kind: NetworkPolicy
+metadata:
+  name: {{ include "dex.fullname" . }}
+  labels:
+    {{- include "dex.labels" . | nindent 4 }}
+spec:
+  policyTypes:
+    - Egress
+    - Ingress
+  podSelector:
+    matchLabels:
+      {{- include "dex.selectorLabels" . | nindent 6 }}
+  ingress:
+    - ports:
+        - port: http
+        {{- if .Values.https.enabled }}
+        - port: https
+        {{- end }}
+        {{- if .Values.grpc.enabled }}
+        - port: grpc
+        {{- end }}
+        - port: telemetry
+  egress:
+    - ports:
+        - port: 53
+          protocol: UDP
+        - port: 53
+          protocol: TCP
+    {{- if .Values.networkPolicy.egressRules }}
+    {{- toYaml .Values.networkPolicy.egressRules | nindent 4 }}
+    {{- end }}
```

I would not specify egress rules by default:

```suggestion
  {{- with .Values.networkPolicy.egressRules }}
  egress:
    {{- toYaml . | nindent 4 }}
  {{- end }}
```

---

@mkilchhofer Thanks for the feedback.
Actually, as soon as any netpol is added to a NS, any other network flow is forbidden.
I thought of dns egress as as a failsafe but I'm fine with change things your way. I'm on it

---

Jup this is true :-) But most users have a global rule in place for DNS such as Calico's `GlobalNetworkPolicy` or Cilium's `CiliumClusterwideNetworkPolicy`.

---

That should be it in fae3976